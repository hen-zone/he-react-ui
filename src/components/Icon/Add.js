// @flow
import React from 'react';

function Add(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 24 24">
      <path d="M10.994 11V6.343a1 1 0 0 1 2 0V11h4.657a1 1 0 0 1 0 2h-4.657v4.657a1 1 0 1 1-2 0V13H6.338a1 1 0 1 1 0-2h4.656z" />
    </svg>
  );
}

Add.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default Add;
