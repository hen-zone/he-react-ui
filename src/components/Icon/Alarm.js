// @flow
import React from 'react';

function Alarm(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 24 24">
      <path d="M19.04 8.2a8.005 8.005 0 0 1-1.18 9.246l1.847 1.847a1 1 0 0 1-1.414 1.414l-1.973-1.972a8.01 8.01 0 0 1-8.64 0l-1.973 1.972a1 1 0 1 1-1.414-1.414l1.847-1.847A8.005 8.005 0 0 1 4.958 8.2a3 3 0 1 1 4.573-3.81 8.034 8.034 0 0 1 4.938 0 3 3 0 1 1 4.572 3.81zm-3.366-.944A6.002 6.002 0 0 0 7.29 8.281a6 6 0 1 0 8.383-1.025zM13 11.5h2.5a1 1 0 0 1 0 2H12a1 1 0 0 1-1-1V9a1 1 0 1 1 2 0v2.5zM6.343 6.343c.41-.41.857-.77 1.336-1.077a1 1 0 0 0-1.523 1.27c.061-.065.123-.13.187-.193zm11.313 0c.064.064.126.129.188.194A.999.999 0 0 0 17 5a1 1 0 0 0-.68.266c.48.308.927.668 1.336 1.077z" />
    </svg>
  );
}

Alarm.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default Alarm;
