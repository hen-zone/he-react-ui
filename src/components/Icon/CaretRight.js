// @flow
import React from 'react';

function CaretRight(props: SVGProps) {
  return (
    <svg {...props} viewBox="0 0 24 24">
      <path d="M2 14.932V7.068a.5.5 0 0 1 .82-.385l4.72 3.933a.5.5 0 0 1 0 .768l-4.72 3.933a.5.5 0 0 1-.82-.385z" />
    </svg>
  );
}

CaretRight.defaultProps = {
  height: 32,
  width: 32,
  fill: 'currentColor',
};

export default CaretRight;
