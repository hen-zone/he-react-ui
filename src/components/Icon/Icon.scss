@import "../../styles/base";

.icon {
  fill: currentColor;
}

.defaultSize {
  height: 32px;
  width: 32px;
}

.teal {
  border-color: $teal_1;
  color: $teal_1;
}

.blue {
  border-color: $sp_blue;
  color: $sp_blue;
}

.green {
  border-color: $sp_green;
  color: $sp_green;
}

.red {
  border-color: $sp_red;
  color: $sp_red;
}

.white {
  border-color: $white;
  color: $white;
}
