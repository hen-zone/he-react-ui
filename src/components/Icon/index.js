// @flow
import classnames from 'classnames';
import React from 'react';
import Add from './Add';
import Alarm from './Alarm';
import Alert from './Alert';
import {
  ArrowLeft,
  ArrowRight,
  ArrowUp,
  ArrowDown,
  CurvedArrow,
} from './Arrows';
import Bookings from './Bookings';
import Calendar from './Calendar';
import CaretRight from './CaretRight';
import {
  CheckBoxChecked,
  CheckBoxUnchecked,
  RadioChecked,
  RadioUnchecked,
} from './CheckBoxes';
import { ChevronDown, ChevronLeft, ChevronRight, ChevronUp } from './Chevrons';
import Clock from './Clock';
import Cross from './Cross';
import Delete from './Delete';
import DropDown from './DropDown';
import Edit from './Edit';
import Ellipsis from './Ellipsis';
import Engage from './Engage';
import HealthEngine from './HealthEngine';
import HealthEngineWithText from './HealthEngineWithText';
import Help from './Help';
import Home from './Home';
import style from './Icon.scss';
import Logout from './Logout';
import Megaphone from './Megaphone';
import Patients from './Patients';
import Search from './Search';
import Settings from './Settings';
import Sync from './Sync';
import Tick from './Tick';
import View from './View';

type Color = 'teal' | 'blue' | 'green' | 'red' | 'white';

function Icon(props: {
  width?: number,
  height?: number,
  color?: Color,
  className?: string,
  name: string,
}) {
  const { width, height, color, className, name, ...restProps } = props;
  const classes = classnames(style.icon, color && style[color], className);
  const childProps = {
    width,
    height,
    className: classes,
    ...restProps,
  };

  switch (name) {
    case 'Add':
      return <Add {...childProps} />;
    case 'Alarm':
      return <Alarm {...childProps} />;
    case 'Alert':
      return <Alert {...childProps} />;
    case 'ArrowLeft':
      return <ArrowLeft {...childProps} />;
    case 'ArrowRight':
      return <ArrowRight {...childProps} />;
    case 'ArrowUp':
      return <ArrowUp {...childProps} />;
    case 'ArrowDown':
      return <ArrowDown {...childProps} />;
    case 'Bookings':
      return <Bookings {...childProps} />;
    case 'Calendar':
      return <Calendar {...childProps} />;
    case 'CaretRight':
      return <CaretRight {...childProps} />;
    case 'CheckBoxChecked':
      return <CheckBoxChecked {...childProps} />;
    case 'CheckBoxUnchecked':
      return <CheckBoxUnchecked {...childProps} />;
    case 'ChevronLeft':
      return <ChevronLeft {...childProps} />;
    case 'ChevronRight':
      return <ChevronRight {...childProps} />;
    case 'ChevronUp':
      return <ChevronUp {...childProps} />;
    case 'ChevronDown':
      return <ChevronDown {...childProps} />;
    case 'Clock':
      return <Clock />;
    case 'CurvedArrow':
      return <CurvedArrow />;
    case 'Cross':
      return <Cross {...childProps} />;
    case 'Delete':
      return <Delete {...childProps} />;
    case 'DropDown':
      return <DropDown {...childProps} />;
    case 'Edit':
      return <Edit {...childProps} />;
    case 'Engage':
      return <Engage {...childProps} />;
    case 'HealthEngine':
      return <HealthEngine {...restProps} />;
    case 'HealthEngineWithText':
      return <HealthEngineWithText />;
    case 'Help':
      return <Help {...childProps} />;
    case 'Home':
      return <Home {...childProps} />;
    case 'Megaphone':
      return <Megaphone {...childProps} />;
    case 'Patients':
      return <Patients {...childProps} />;
    case 'Ellipsis':
      return <Ellipsis {...childProps} />;
    case 'Logout':
      return <Logout {...childProps} />;
    case 'RadioChecked':
      return <RadioChecked {...childProps} />;
    case 'RadioUnchecked':
      return <RadioUnchecked {...childProps} />;
    case 'Search':
      return <Search {...childProps} />;
    case 'Settings':
      return <Settings {...childProps} />;
    case 'Sync':
      return <Sync {...childProps} />;
    case 'Tick':
      return <Tick {...childProps} />;
    case 'View':
      return <View {...childProps} />;
    default:
      return <Cross {...childProps} />;
  }
}

Icon.defaultProps = {
  name: 'Cross',
};

export default Icon;
