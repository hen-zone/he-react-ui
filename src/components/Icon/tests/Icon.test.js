// @flow
import testFixture from '../../../../config/testFixture';
import Icon from '../fixtures/Icon.fixture';

testFixture(Icon);
