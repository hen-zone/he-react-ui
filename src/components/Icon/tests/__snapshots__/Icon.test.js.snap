// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> rendered correctly with Add fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Alarm fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Alert fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Arr fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowLeft fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowLeft fixture 2`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowRight fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ArrowUp fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Bookings fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Calendar fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with CaretRight fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with CheckBoxChecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with CheckBoxUnchecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronLeft fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronRight fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with ChevronUp fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Clock fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Cross fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with CurvedArrow fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Delete fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with DropDown fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Edit fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Ellipsis fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Engage fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with HealthEngine fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with HealthEngineWithText fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Help fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Home fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Logout fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Megaphone fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Patients fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with RadioChecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with RadioUnchecked fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Search fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Settings fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Sync fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with Tick fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;

exports[`<Icon /> rendered correctly with View fixture 1`] = `
<Icon
  name="ArrowUp"
>
  <ArrowUp
    className="icon"
    fill="currentColor"
    height={32}
    width={32}
  >
    <svg
      className="icon"
      fill="currentColor"
      height={32}
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13 5.414V21a1 1 0 1 1-2 0V5.414L6.707 9.707a1 1 0 1 1-1.414-1.414l6-6a1 1 0 0 1 1.414 0l6 6a1 1 0 0 1-1.414 1.414L13 5.414z"
      />
    </svg>
  </ArrowUp>
</Icon>
`;
